### PR TITLE
Merge of THG work on post_open_fix branch

### PR DESCRIPTION
**Pull request by derobins**
Includes:
* Recent merges to develop
* H5Tconv.c change from last NERSC sync (lots of churn on that vlen reference fix from Neil)
* CMake fixes

### CODE REVIEW

**Comment by qkoziol** [comment]
*blah*   @derobins - would you mind rebasing?

**Comment by derobins** [comment]
> _blah_ @derobins - would you mind rebasing?

Blah is right. I think the changes were also merged to develop independently so I see about a dozen files with conflicts. I'll fix that up tonight so it's ready for tomorrow.

**Comment by derobins** [comment]
The PR should be back in sync w/ THG's develop branch

File: hl/tools/h5watch/Makefile.am [hunk 1]
```diff
@@ -37,8 +36,8 @@ if BUILD_TESTS_CONDITIONAL
   AM_CPPFLAGS+=-I$(top_srcdir)/hl/test
   TEST_SCRIPT=testh5watch.sh
   check_SCRIPTS=$(TEST_SCRIPT)
-  SCRIPT_DEPEND=extend_dset$(EXEEXT) h5watch$(EXEEXT)
-  noinst_PROGRAMS+=h5watchgentest extend_dset
+  SCRIPT_DEPEND=swmr_check_compat_vfd$(EXEEXT) extend_dset$(EXEEXT) h5watch$(EXEEXT)
```
Comment (qkoziol) [inline hl/tools/h5watch/Makefile.am:39]: I don't think the swmr_check_compat_vfd should be here, but I'll double-check when I test.

**Review by qkoziol — APPROVED** [review]

**Pull request merged**